src/Shared/CertificateGeneration/WindowsCertificateManager.cs

```diff
@@ -41,8 +41,12 @@ protected override bool IsExportable(X509Certificate2 c)
 
     internal override CheckCertificateStateResult CheckCertificateState(X509Certificate2 candidate, bool interactive)
     {
-        // Return true as we don't perform any check.
-        return new CheckCertificateStateResult(true, null);
+        if (IsTrusted(candidate))
+        {
+            return new CheckCertificateStateResult(true, null);
+        }
+
+        return new CheckCertificateStateResult(false, "The ASP.NET Core developer certificate is not trusted.");
```

Comment:
Why are we checking this here?

`CheckCertificateState` was here to detect when we needed to take action to fix the certificate on Mac OS (which we might need to take again)